Framework/Interaction/Tax.php

```diff
@@ -870,4 +882,22 @@ protected function areTimesDifferentDays($timeA, $timeB, $storeId)
 
         return $dateA->format('Y-m-d') != $dateB->format('Y-m-d');
     }
+
+    /**
+     * Return whether product is calculated or not
+     *
+     * @param $item
+     * @return bool
+     */
+    protected function isProductCalculated($item) {
+        if (method_exists($item, 'isChildrenCalculated') && method_exists($item, 'getParentItem')) {
+            if ($item->isChildrenCalculated() && !$item->getParentItem()) {
+                return true;
+            }
+            if (!$item->isChildrenCalculated() && $item->getParentItem()) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
I'm concerned about defaulting to "false". Are you certain that all product types (simple, configurable, bundled, virtual, downloadable, etc) that are supposed to get taxes have `isChildrenCalculated` _and_ `getParentItem` methods?